code/controllers/mc/admin.dm

```diff
@@ -53,6 +63,35 @@ INITIALIZE_IMMEDIATE(/obj/statclick)
 
 	message_admins("Admin [key_name_admin(usr)] has restarted the [controller] controller.")
 
+/client/proc/debug_controller()
+	set category = "Debug.Controllers"
+	set name = "Debug Controller"
+	set desc = "Debug the various periodic loop controllers for the game (be careful!)"
+
+	if(!admin_holder)
+		return
+
+	var/list/controllers = list()
+	var/list/controller_choices = list()
+
+	for (var/datum/controller/controller in world)
+		if (istype(controller, /datum/controller/subsystem))
+			continue
+		controllers["[controller] (controller.type)"] = controller //we use an associated list to ensure clients can't hold references to controllers
+		controller_choices += "[controller] (controller.type)"
+
+	var/datum/controller/controller_string = input("Select controller to debug", "Debug Controller") as null|anything in controller_choices
```

Comment:
Tgui input list?

Reply:
This is probably intended since you'd want to eg. debug TGUI controller due to issues - the other way is to use Statpanel Status entries but that is also controller backed and might also fail. Verbs are always available via command bar even if everything else fails

Can replace it if you think it's desirable to have the menu there

Reply:
fair, can keep old list